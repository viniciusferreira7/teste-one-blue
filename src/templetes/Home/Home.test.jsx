import { render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { Home } from '.';

describe('<Home/>', () => {
  it('should render children components', async () => {
    await act(async () => render(<Home />));
    expect(document.querySelector('form')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Entrar' })).toBeInTheDocument();
  });

  it('should match snapshot', async () => {
    await act(async () => render(<Home />));
    expect(document.querySelector('form')).toBeInTheDocument();
    expect(
      screen.getByRole('button', { name: 'Entrar' }).parentElement,
    ).toMatchSnapshot();
  });
});
