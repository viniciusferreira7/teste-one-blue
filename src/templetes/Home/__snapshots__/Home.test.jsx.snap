// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> should match snapshot 1`] = `
<div
  class="sc-bczRLJ fgVWkE"
>
  <form
    class="sc-iBkjds jnGyuj"
  >
    <h1>
      Cadastre-se
    </h1>
    <p
      class="sc-gKXOVf hjAfTq"
    >
      Necessário preencher o campo de senha
    </p>
    <input
      class="sc-dkzDqf iXafZv"
      name="name"
      placeholder="Digite seu nome de usuário"
      required=""
      type="text"
    />
    <input
      class="sc-hKMtZM cLivwc"
      name="password"
      placeholder="Digite sua senha"
      required=""
      type="password"
    />
    <button
      class="sc-eCYdqJ fSkBnH"
    >
      Cadastrar
    </button>
  </form>
  <button>
    Entrar
  </button>
</div>
`;
